### DISCUSSION TITLE
Interested in using kitty in the same way fzf-tmux works

### DISCUSSION THREAD

**Discussion by megalithic** [junegunn/fzf#2760] [Q&A] [answered]
Hi! 

I'm looking at ways to use kitty, as a replacement, in many ways, to tmux. One thing that tmux allows, which is so helpful (and fzf-tmux makes possible) is.. fzf in a quick pane. Kitty has _something_ similar, in way of different types of targets for kitty, one being an overlay, which is a "quick" full size popup. 

So, my question is, has anyone gotten fzf-tmux-like functionality via kitty and it's overlay launch type?

**Comment by duganchen** [comment, marked as answer]
This works on my Mac. It works both in a pipeline (>ls | fzf-kitty) and not (>fzf-kitty). It involves two scripts.

The first is *_fzf-kitty-to*, which is for the fzf-kitty script to call:

    #!/bin/bash

    # https://github.com/junegunn/fzf/issues/2481#issuecomment-839591602
    tail_app="tail"
    if [[ "$(uname)" == "Darwin" ]]; then
        tail_app=gtail
    fi

    if [[ -f "$INFILE" ]]; then
        $tail_app -f "$INFILE" | fzf "$@" >"$OUTPIPE"
    else
        fzf "$@" >"$OUTPIPE"
    fi
    

And the actual *fzf-kitty* script:

    #!/bin/sh

    outpipe=$(mktemp -u)
    mkfifo -m 600 "$outpipe"

    if [ ! -t 0 ]; then
        infile=$(mktemp /tmp/fzf-kitty.XXXXXX)
        while read -r line; do
            echo "$line" >>"$infile"
        done
    fi

    kitty @launch --cwd "$PWD" --env "PATH=$PATH" --env "INFILE=$infile" --env "OUTPIPE=$outpipe" _fzf-kitty-to "$@" >/dev/null

    read -r line <"$outpipe"

    if [ ! -t 0 ]; then
        rm "$infile"
    fi
    rm "$outpipe"
    echo "$line"
